lightning/src/chain/chaininterface.rs

```diff
@@ -65,6 +65,13 @@ pub trait ChainWatchInterface: Sync + Send {
 }
 
 /// An interface to send a transaction to the Bitcoin network.
+///
+/// We may pass non-final transaction (like local HTLC-timeout tx)
+/// in case of channel force-closure due to an offchain anomaly. To warn
+/// being bounce off of every network mempools, implementation MUST
+/// wait until transaction finalization before p2p announcement. To avoid
```

Comment:
I dont think there's any specific harm in broadcasting them beforehand, as long as they are rebroadcast on every block.

Reply:
And that's exactly I want to avoid, if transaction isn't final that's a privacy leak to rebroadcast at every block?